static/app/utils/profiling/renderers/gridRenderer.tsx

```diff
@@ -0,0 +1,138 @@
+import {mat3, vec2} from 'gl-matrix';
+
+import {FlamegraphTheme} from '../flamegraph/FlamegraphTheme';
+import {getContext, measureText, Rect} from '../gl/utils';
+
+export function getIntervalTimeAtX(configToPhysicalSpace: mat3, x: number): number {
+  const logicalToPhysical = mat3.fromScaling(
+    mat3.create(),
+    vec2.fromValues(window.devicePixelRatio ?? 1, window.devicePixelRatio ?? 1)
+  );
+  const physicalToConfigSpace = mat3.invert(mat3.create(), configToPhysicalSpace);
+
+  const logicalToConfigSpace = mat3.multiply(
+    mat3.create(),
+    physicalToConfigSpace,
+    logicalToPhysical
+  );
+
+  return vec2.transformMat3(
+    vec2.create(),
+    vec2.fromValues(x, 1),
+    logicalToConfigSpace
+  )[0];
```

Comment:
Nit: Since you're only interested in the result of applying the transformation to `x`, we could just directly compute with something like

```javascript
logicalToConfigSpace[0] * x + logicalToConfigSpace[3] + logicalToConfigSpace[6]
```

Reply:
👍🏼 I just noticed that vec2 does some nice rounding of floating points for us, so we'll need to handle it a bit, but I like it